packages/apps/src/app.oauth.ts

```diff
@@ -99,3 +100,5 @@ export async function onVerifyState(
     return { status: 412 };
   }
 }
+export type OnTokenExchange<TPlugin extends IPlugin> = typeof onTokenExchange<TPlugin>;
```

Comment:
it would be nice to organize types a bit more consciously so that in the long term we don't have dangling types like this all over the place.

Reply:
Yeah that's totally fair. let me see if i can clean it up.

Reply:
Okay I had to do this to avoid the explicit type creation:
```ts
this.on('signin.token-exchange', (...args) => this.onTokenExchange(...args));
this.on('signin.verify-state', (...args) => this.onVerifyState(...args));
```
Not a huge deal imo, but lmk if you disagree. Basically typescript doesn't know what the value of `this` is when i'm binding it unless I explicitly do 
```ts
onTokenExchange: OnTokenExchange<TPlugin> = onTokenExchange
```